extensions/amp-serviceworker-installation/0.1/amp-serviceworker-installation.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Layout} from '../../../src/layout';
+import {parseUrl, assertHttpsUrl} from '../../../src/url';
+import {getMode} from '../../../src/mode';
+
+/**
+ * Implements custom element: <amp-serviceworker-installation>
+ * for installation of ServiceWorkers owned by the publisher
+ * of the current page.
+ */
+class AmpServiceWorkerInstallation extends AMP.BaseElement {
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.NODISPLAY;
+  }
+
+  /** @override */
+  buildCallback() {
+    const win = this.getWin();
+    if (!('serviceWorker' in win.navigator)) {
+      return;
+    }
+    const src = this.element.getAttribute('src');
+    assertHttpsUrl(src, this.element);
+
+    if (originMatches(win.location.href, src)) {
+      install(this.getWin(), src);
+    } else {
+      if (getMode().development) {
+        console./* OK */warn(
```

Comment:
Definitely not for now. This triggered me asking about it. Our current logging infra is OK for us, but nothing that a non-core-AMP engineer should ever see.

I'm super +1 on having shared infra for logging messages to users.